evm_arithmetization/src/witness/transition.rs

```diff
@@ -308,7 +308,9 @@ pub(crate) fn log_kernel_instruction<F: RichField, S: State<F>>(state: &mut S, o
         ),
     );
 
-    assert!(pc < KERNEL.code.len(), "Kernel PC is out of range: {}", pc);
+    let kernel_code =
+        &state.get_generation_state().memory.contexts[0].segments[Segment::Code.unscale()].content;
+    assert!(pc < kernel_code.len(), "Kernel PC is out of range: {}", pc);
```

Comment:
nonblocking/education: I'd use `assert2::assert!` which would print the left and right values in the panic